tools/libboard/board_dc/board_dc.c

```diff
@@ -130,8 +130,7 @@ fpga_result parse_fw_ver(char *buf, char *fw_ver, size_t len)
 
 	errno = 0;
 	var = strtoul(buf, &endptr, 16);
-	if (var == 0  &&
-		errno != 0) {
+	if (endptr && *endptr == '\0') {
```

Comment:
Whoops, this is the success case, so you would need:
```c
if (!endptr || *endptr != '\0') {
```

Reply:
fixed